Update description for exercism.io (#20)
BostonPython/about: resources.md

**NOTE**: This is a work-in-progress. The resources listed have only been lightly curated, and there are likely other great ones that should be on this list. We welcome [suggestions](#edit-page).

## Places to ask for help when you're stuck
- the Boston Python [Slack workspace](https://about.bostonpython.com/slack)
- [Dream In Code](https://www.dreamincode.net/) is a beginner-friendly discussion-oriented programming forum. 
- [Stack Overflow](https://stackoverflow.com/) is a good place to find quick answers to specific technical questions, but is usually not a good place for involved discussion about logic or problem-solving. 

## Python-specific exercises

> - [exercism](https://exercism.io/): Crowd-sourced <q>code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Installing their command-line interface can be a little complicated, but after that the submission system is smoother than many online exercise portals. Problems are submitted and maintained by volunteers, as a result some problem descriptions may be incomplete or confusing. (see above for 

Reviewer: Nice positive spin! I think "exercises" is more consistent, and less ambiguous, than "problems".

```suggestion
- [exercism](https://exercism.io/): Crowd-sourced <q>code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Installing their command-line interface can be a little complicated, but after that the submission system is smoother than many online exercise portals. Exercises are submitted and maintained by volunteers, as a result some exercise descriptions may be incomplete or confusing.
```

## Python-specific exercises

> - [exercism](https://exercism.io/): Crowd-sourced <q>code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Installing their command-line interface can be a little complicated, but after that the submission system is smoother than many online exercise portals. Beginners should be aware that some problem descriptions may be incomplete or confusing. Feel free to ask for help on the slack if a problem seems unclear!

Reviewer: ```suggestion
- [exercism](https://exercism.io/): Crowd-sourced <q>code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Installing their command-line interface can be a little complicated, but after that the submission system is smoother than many online exercise portals. Some exercise descriptions may be incomplete or confusing.
```

Thanks for taking my suggestion; here's another one. ;)

I like the intention of "ask for help", but it could apply to any of these resources, and not all the readers will know what "the slack" is.

How about adding a sentence like this after the first paragraph of these page?

```
If you need help with a specific exercise, feel free to ask on our [Slack workspace](https://about.bostonpython.com/slack).
```

> - [exercism](https://exercism.io/): <q>Code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Exercism is free and open-source, which is nice. It requires installation of their submission software, which is a slightly cumbersome process, but once installed the submission system is smoother than many online exercise portals. Problems are not always well-specified, and it may at times be difficult to understand exactly what is required.

Reviewer: In the interest of keeping this short and inviting, what about something like this?

```suggestion
- [exercism](https://exercism.io/): Crowd-sourced <q>code practice and mentorship for everyone</q>, with over 100 [Python-specific exercises](https://exercism.io/tracks/python). Installing their command-line interface can be <q>a little complicated</q> , but after that the submission system is smoother than many online exercise portals.
```

If you feel strongly about "Problems are not always well-specified", can that be expressed more positively/succinctly? I'd also probably use "Exercises" instead of "Problems". And I'm curious: can you share an example or two of such exercises?

Author: One example of unclear spec: https://exercism.io/my/solutions/4f3ea117875143d5ad8f8cf2fa892642

The problem specification, in full, is "Output the lyrics to 'The Twelve Days of Christmas'.", followed by the lyrics of the last verse of that song. ("in full" meaning the unique content - there is a lot of boilerplate repeated on every problem)

If you look at the tests, it's possible to work out that what they want is a function that recites verses of the song from some `start_verse` to some `end_verse`, inclusive, but I would expect this information to be included in the problem spec.

Author: I'm not sure how to express this more positively or succinctly. Some of the problem descriptions I looked at were fine, others were confusing or incomplete. 
I do think it's worth raising this flag, because encountering an unexpectedly confusing problem is likely to be demoralizing for some beginners.
